### PR TITLE
QuarkusComponentTest: convenient way of mocking interceptors

### LINKED ISSUE #34086: QuarkusComponentTest: offer a convenient way of mocking interceptors
### Description

Currently, if an interceptor binding is declared in a tested component and we would like to reflect the binding in the test, then an mock interceptor class must be registered as an additional component, e.g. something like:

```java
@QuarkusComponentTest(SimpleInterceptor.class)
public class InterceptorMockingTest {

    @Inject
    TheComponent theComponent;

    @Test
    public void testPing() {
        assertEquals("ok", theComponent.ping());
    }

    @Singleton
    static class TheComponent {

        @SimpleBinding
        String ping() {
            return "true";
        }

    }
    @Priority(1)
    @SimpleBinding
    @Interceptor
    static class SimpleInterceptor {

        @AroundInvoke
        Object aroundInvoke(InvocationContext context) throws Exception {
            return "ok";
        }
    }
}
```

Moreover, this interceptor class may cause problems in regular `@QuarkusTest`s, because all classes in `src/test/java` are discovered by default. So the [%test.quarkus.arc.exclude-types](https://quarkus.io/guides/cdi-reference#quarkus-arc_quarkus.arc.exclude-types) config property would need to be used in order to exclude the interceptor class from discovery.

### Implementation ideas

I was thinking that we could support "interceptor methods" declared direcly on the test class.

```java
@QuarkusComponentTest
public class InterceptorMockingTest {

    @Inject
    TheComponent theComponent;

    @Test
    public void testPing() {
        assertEquals("ok", theComponent.ping());
    }

    @Singleton
    static class TheComponent {

        @SimpleBinding
        String ping() {
            return "true";
        }

    }

    @SimpleBinding
    @AroundInvoke
    Object aroundSimpleBinding(InvocationContext context) throws Exception {
            return "ok";
    }
}
```

That would of course require support of synthetic interceptors in ArC (not supported at the moment)

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #34086
- [x] Support `@Priority` declared on a test interceptor method
- [x] Documentation

### CODE REVIEW

File: test-framework/junit5-component/src/main/java/io/quarkus/test/component/InterceptorMethodCreator.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    static final String CREATE_KEY = "createKey";
+
+    private static final Map<String, Function<SyntheticCreationalContext<?>, InterceptFunction>> createFunctions = new HashMap<>();
+
+    @Override
+    public InterceptFunction create(SyntheticCreationalContext<Object> context) {
+        Object createKey = context.getParams().get(CREATE_KEY);
+        if (createKey != null) {
+            Function<SyntheticCreationalContext<?>, InterceptFunction> createFun = createFunctions.get(createKey.toString());
```
Comment (Ladicek) [inline test-framework/junit5-component/src/main/java/io/quarkus/test/component/InterceptorMethodCreator.java:20]: I don't think `toString()` here is necessary, `Map.get()` takes an `Object` (and not a `K`).

Comment (Ladicek) [reply test-framework/junit5-component/src/main/java/io/quarkus/test/component/InterceptorMethodCreator.java:20]: Though I see it's the same in `MockBeanCreator`, where I totally missed it :-)

It's a cosmetic thing, anyway.

**Review by Ladicek — APPROVED** [review]
To be honest, I don't really like this way of declaring interceptors, but the code LGTM.

**Comment by mkouba** [comment]
> To be honest, I don't really like this way of declaring interceptors, but the code LGTM.

I know, you told me already in https://github.com/quarkusio/quarkus/issues/34086. That's why I was working on https://github.com/quarkusio/quarkus/pull/34127 first.

**Pull request merged**